storage/storage_dest.go

```diff
@@ -365,13 +365,15 @@ func (s *storageImageDestination) tryReusingBlobAsPending(blobDigest digest.Dige
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	if options.SrcRef != nil {
+	if options.SrcRef != nil && options.TOCDigest != "" && options.TOCDigest.Validate() == nil {
```

Comment:
Please rebase – the `Validate` check now happens before this line.

Reply:
rebased